source/module_lr/lr_spectrum.cpp

```diff
@@ -53,7 +53,7 @@ void LR::LR_Spectrum<double>::oscillator_strength(Grid_Driver& gd, const std::ve
 
             // 2. transition density
             double** rho_trans;
-            LR_Util::new_p2(rho_trans, 1, this->rho_basis.nrxx);
+            LR_Util::new_pointer_dim2(rho_trans, 1, this->rho_basis.nrxx);
```

Comment:
I would prefer the name "_allocate_2order_nested_ptr", reasons are:
1. it is a function only for make your development easy, so use "_" leading, it is not a quite formal function
2. use "allocate" instead of "new", because "allocate" is human language, "new" is a technical keywords. What if a Python/Fortran programmer read your code? He/she cannot quickly realize what the "new" means
3. the "dim2" is not clear, because you cannot assume a pointer has two dimension, instead, it is the operation allocate memory for a nested pointer in-one-shot

Thanks.

Reply:
Please also consider my suggestion in the case of dual function `delete_pointer_dim2`